lib/src/test/java/io/ably/lib/test/realtime/RealtimeChannelTest.java

```diff
@@ -1126,4 +1127,142 @@ public void detach_on_clean_connection_preserves_channel() {
 				ably.close();
 		}
 	}
+
+
+	@Test
+	public void channel_state_on_connection_suspended() {
+		AblyRealtime ably = null;
+		try {
+			TestVars testVars = Setup.getTestVars();
+			ClientOptions opts = testVars.createOptions(testVars.keys[0].keyStr);
+
+			ably = new AblyRealtime(opts);
+
+			/* wait until connected */
+			(new ConnectionWaiter(ably.connection)).waitFor(ConnectionState.connected);
+			assertEquals("Verify connected state reached", ably.connection.state, ConnectionState.connected);
+
+			/* create a channel and attach */
+			final String channelName = "test_state_channel";
+			final Channel channel = ably.channels.get(channelName);
+			channel.attach();
+
+			ChannelWaiter channelWaiter = new ChannelWaiter(channel);
+			channelWaiter.waitFor(ChannelState.attached);
+			assertEquals("Verify attached state reached", channel.state, ChannelState.attached);
+
+			/* switch to suspended state */
+			ably.connection.connectionManager.requestState(ConnectionState.suspended);
```

Comment:
I'm inclined to think that this should be `notifyState()` - I thought `requestState()` is for the situation where the `ConnectionManager` needs to take some action to put it into that state - eg it calls `connectImpl()` in response to a call to `requestState("connecting")`